cpmonitor/views.py

```diff
@@ -16,9 +16,33 @@
 from .models import City, ExecutionStatus, Task
 
 
-def _get_children(city, node=None):
+def _calculate_summary(node):
+    """calculate summarized status for a give node or a whole city"""
+
     statuses = {s.value: s for s in ExecutionStatus}
+    if isinstance(node, City):
+        subtasks = [
+            t
+            for r in Task.get_root_nodes()
```

Comment:
I guess, it would be good to filter for the city, here:
```python
            for r in Task.get_root_nodes().filter(city=node)
```
(untested)

Reply:
Will do this and merge.

Reply:
Not only good, but necessary. Anything else would be a bug. Thanks for spotting and fixing that.